resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileStore.java

```diff
@@ -29,13 +38,25 @@
  */
 public class NullFileStore extends FileStore {
 	private final IPath path;
+	protected final IFileSystem fileSystem;
 
 	/**
 	 * Creates a null file store
 	 * @param path The path of the file in this store
 	 */
-	public NullFileStore(IPath path) {
+	public NullFileStore(IPath path, IFileSystem fileSystem) {
```

Comment:
Why adding an argument / potentially breaking clients that might have used it for years (and yes, this is not API)?

However, I've pushed now second commit that uses `EFS.getNullFileSystem()` https://github.com/eclipse-platform/eclipse.platform/pull/1137.

Reply:
> Why adding an argument / potentially breaking clients that might have used it for years (and yes, this is not API)?

Because it is not API and it is obviously misusing that, I think it is acceptable this for a extender but not for a public constructor. Do you know of any client that can be affected?

Reply:
I now added a "compatibility" constructor ...

Reply:
> But from my experience, clients may use everything that is public and is there for nearly 20 years, so if possible I would always avoid breaking them for no obvious reason.

That's why we mark it internal so clients are aware they can break anytime. And as we can see here this even introduced a bug in BndTools, as the misuses the `NullFileStore` for a custom implementation that is completely unrelated, anyways with the fallback everything should now be fine.